orderer/consensus/etcdraft/node.go

```diff
@@ -151,6 +151,11 @@ func (n *node) run(campaign bool) {
 				n.chain.snapC <- &rd.Snapshot
 			}
 
+			// skip empty apply
```

Comment:
> The chain's apply channel is processed by another goroutine, so it might take the apply from the channel, be preempted, and then this goroutine will reach line 168. So in a sense, there exists a scheduling in the new code form that is equivalent to the old code form, don't you agree?
I completely agree.  It is not deterministic due to thread/go rountine scheduling/context switch.  

Since, now we are moving the subscription notification after the chain applied the state changes. it is now streamlined.

Reply:
But now we are firing the subscription twice, no?

In the chain, line 164, and in the node, line 800.

Reply:
sorry, it was mistake while adding the changes into the commit. I corrected it now.

Reply:
Looks good, let's get some more eyes on this. 

@tock-ibm would you please look at this change?

Reply:
nit: 
The comment line 1883 in the chain_test.go should be:
// remove first consenter, which is the leader

Otherwise, looks good to me.